mmeval/metrics/niqe.py

```diff
@@ -0,0 +1,462 @@
+        **kwargs: Keyword parameters passed to :class:`BaseMetric`.
+
+    Examples:
+
+        >>> from mmeval import NIQE
+        >>> import numpy as np
+        >>>
+        >>> niqe = NIQE(input_order='CHW', convert_to='Y', channel_order='rgb')
+        >>> preds = np.random.randint(0, 255, size=(3, 32, 32))
+        >>> niqe(preds)  # doctest: +ELLIPSIS
+        {'niqe': ...}
+    """
+
+    def __init__(self,
+                 crop_border: int = 0,
+                 input_order: str = 'CHW',
+                 convert_to: str = 'gray',
+                 channel_order: str = 'rgb',
+                 **kwargs) -> None:
+        super().__init__(**kwargs)
+
+        assert input_order.upper() in [
+            'CHW', 'HWC'
+        ], (f'Wrong input_order {input_order}. Supported input_orders are '
+            '"HWC" and "CHW"')
+        assert convert_to in [
+            'y', 'gray', 'Y'
+        ], ('Only support gray image, '
+            "``convert_to`` should be selected from ['y', 'gray']")
+
+        self.crop_border = crop_border
+        self.input_order = input_order
+        self.convert_to = convert_to
+        self.channel_order = channel_order
+
+    def add(self, predictions: Sequence[np.ndarray], channel_order: Optional[str] = None) -> None:  # type: ignore # yapf: disable # noqa: E501
+        """Add NIQE score of batch to ``self._results``
+
+        Args:
+            predictions (Sequence[np.ndarray]): Predictions of the model.
+            channel_order (Optional[str]): The channel order of the input
+                samples. If not passed, will set as :attr:`self.channel_order`.
+                Defaults to None.
+        """
+
+        # we use the official params estimated from the pristine dataset.
+        niqe_pris_params = np.load(
+            os.path.join(os.path.dirname(__file__), 'niqe_pris_params.npz'))
+        mu_pris_param = niqe_pris_params['mu_pris_param']
+        cov_pris_param = niqe_pris_params['cov_pris_param']
+        gaussian_window = niqe_pris_params['gaussian_window']
+
+        if channel_order is None:
+            channel_order = self.channel_order
+
+        for prediction in predictions:
+
+            if len(prediction.shape) <= 3:
+                result = self.compute_niqe(prediction, channel_order,
+                                           mu_pris_param, cov_pris_param,
+                                           gaussian_window)
+            else:
+                result_sum = 0
+                for i in range(prediction.shape[0]):
+                    result_sum += self.compute_niqe(prediction[i],
+                                                    channel_order,
+                                                    mu_pris_param,
+                                                    cov_pris_param,
+                                                    gaussian_window)
+                result = result_sum / prediction.shape[0]
+
+            self._results.append(result)
```

Comment:
What does `prediction.shape[0]` mean, the `batch_size`?

If the `batch_size` of each `predictions` is not equal (e.g. the last batch of dataloader has smaller `batch_size`), this implementation might be wrong. Maybe we should make `result` a tuple of `(sum, count)` and do reduce in `compute_metric()`?

Reply:
`prediction.shape[0]` is the number of video frames. This implementation should be correct.